octoprint_mrbeam/static/js/convert.js

```diff
@@ -370,7 +370,7 @@ $(function () {
                     var fm = self.filteredMaterials();
                     for (var i = 0; i < fm.length; i++) {
                         var my_material = fm[i];
-                        if (my_material.name === new_material.name) {
+                        if (my_material.key === new_material.key) {
```

Comment:
This is not the reference key in the material mapping. I just discovered that the `key` attribute exists for the material settings but I have no idea what it is for.

Reply:
This turned out to be faulty change. It's fixed now.